src/realm/table.hpp

```diff
@@ -1641,6 +1643,19 @@ inline bool Table::has_shared_type() const noexcept
     return !m_top.is_attached();
 }
 
+inline void Table::verify_column(size_t col_ndx, const ColumnBase* col) const
+{
+    if (col_ndx < m_cols.size()) {
+        if (m_cols[col_ndx] == col)
```

Comment:
I assume that this is intended as an optimization for the case where no columns have been moved or removed?